Targets/Mashow/Sources/DrinkSelection/View/DrinkTypeViewController.swift

```diff
@@ -0,0 +1,179 @@
+        button.layer.cornerRadius = 20
+        button.layer.masksToBounds = true
+        
+        let view = UIStackView()
+        view.axis = .horizontal
+        view.spacing = 4
+        view.alignment = .center
+        let label = UILabel()
+        label.font = .pretendard(size: 16, weight: .bold)
+        var icon = UIImageView(image: UIImage(systemName: "plus"))
+        
+        let gradient = CAGradientLayer()
+        gradient.frame = button.bounds
+        gradient.colors = [
+            UIColor.hex(drinkType.colorHexValues[0]).cgColor,
+            UIColor.hex(drinkType.colorHexValues[1]).cgColor
+        ]
+        gradient.startPoint = CGPoint(x: 0.0, y: 0.0)
+        gradient.endPoint = CGPoint(x: 1.0, y: 0.0)
+        button.layer.insertSublayer(gradient, at: 0)
+        label.text = "추가됨"
+        label.textColor = UIColor.hex("434343")
+        icon = UIImageView(image: UIImage(systemName: "checkmark"))
+        icon.tintColor = UIColor.hex("434343")
+        
+        view.addArrangedSubview(label)
+        view.addArrangedSubview(icon)
+        button.addSubview(view)
+        view.bounds = button.bounds
+        view.snp.makeConstraints { make in
+            make.center.equalTo(button)
+        }
+        button.isEnabled = false
+        return button
+    }()
+    
+    private lazy var drinkBottleImageView: UIImageView = {
+        let image = UIImage(named: drinkType.rawValue + "Bottle")
+        let imageView = UIImageView(image: image)
+        imageView.contentMode = .scaleAspectFill
+        return imageView
+    }()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        bind()
+        setupLayouts()
+        setupHandlers()
+        viewModel.state.currentType.send(drinkType) // FIXME: - 드래그 조금만 해도 호출돼서 버그있음
+    }
+}
+
+private extension DrinkTypeViewController {
+    
+    private func bind() {
+        viewModel.state.addedTypesPublisher
+            .receive(on: DispatchQueue.main)
+            .sink { [weak self] addedTypes in
+                guard let self = self else { return }
+                let isTypeAdded = addedTypes.contains(self.drinkType)
+                UIView.transition(with: view, duration: 0.5, options: .transitionCrossDissolve) {
+                    self.addDrinkTypeButton.isHidden = isTypeAdded
+                    self.addedDrinkTypeButton.isHidden = !isTypeAdded
+                }
+                if !isTypeAdded, addedTypes.count < 3 {
+                    self.addDrinkTypeButton.isEnabled = true
+                    self.addDrinkTypeButton.layer.opacity = 1.0
+                }
+            }
+            .store(in: &cancellables)
+    }
+    
+    private func setupLayouts() {
+        view.addSubview(typeTitleImageView)
+        typeTitleImageView.snp.makeConstraints { make in
+            make.centerX.equalToSuperview()
+            make.top.equalTo(view.safeAreaLayoutGuide)
```

Comment:
얘가 child 뷰 컨트롤러라 top constraint을 결정할 권한은 없는 것 같긴 하지만.. (ex. 부모 뷰컨에서 top을 화면의 센터로 잡으면..?)
```suggestion
            make.top.equalToSuperView()
```
```swift
// 부모 뷰컨에서
view.addSubview(pageViewController.view)
pageViewController.view.snp.makeConstraints { make in
    make.top.left.trailing.equalTo(view.safeAreaLayoutGuide)
    make.bottom.equalToSuperview()
  }
```
사실 원칙대로면 이런식으로 부모 뷰컨에서 `safeAreaLayoutGuide`를 잡아주는게 맞긴 합니당.
근데 딱히 재사용이 되는 뷰는 아니라 그냥 놔둬도 될듯?

Reply:
흐흐아암 safearea로 안잡아주면 넘어가버리는데 ... 이렇게 되는게 이상한거야 ??

<img width="340" alt="스크린샷 2024-09-03 오후 8 10 29" src="https://github.com/user-attachments/assets/2ae76c99-eb74-434d-a657-93c401bc8ccb">